### PR TITLE
Update some torchscript tests after #24505

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

Need to update the logic in some torchscript tests after #24505

### CODE REVIEW

File: tests/models/blip/test_modeling_blip.py [hunk 1]
```diff
@@ -496,6 +496,15 @@ def _create_and_check_torchscript(self, config, inputs_dict):
             model_state_dict = model.state_dict()
             loaded_model_state_dict = loaded_model.state_dict()
 
+            non_persistent_buffers = {}
```
Comment (sgugger) [inline tests/models/blip/test_modeling_blip.py:499]: You don't care about the values here, so let's just do a set.

File: tests/models/blip/test_modeling_blip.py [hunk 2]
```diff
@@ -496,6 +496,15 @@ def _create_and_check_torchscript(self, config, inputs_dict):
...
+            for key in loaded_model_state_dict.keys():
+                if key not in model_state_dict.keys():
+                    non_persistent_buffers[key] = loaded_model_state_dict[key]
```
Comment (sgugger) [inline tests/models/blip/test_modeling_blip.py:502]:
```suggestion
                    non_persistent_buffers.add(key)
```

**Review by sgugger — COMMENTED** [review]
Thanks for updating those! Left a comment to replicate across the tests.

**Comment by ydshieh** [comment]
@sgugger Thanks for the suggestion. I actually copied from the common test file. However, there is 

```python
            model_buffers = list(model.buffers())
            for non_persistent_buffer in non_persistent_buffers.values():
                found_buffer = False
                for i, model_buffer in enumerate(model_buffers):
                    if torch.equal(non_persistent_buffer, model_buffer):
                        found_buffer = True
                        break

                self.assertTrue(found_buffer)
                model_buffers.pop(i)
```
there (which I didn't see when working on this PR). This uses the `values`.

So I am going to copy the above block (and therefore keeping using dict). Let me know if you have other opinions instead.

**Comment by sgugger** [comment]
If you end up using the values, no worries.

**Review by sgugger — APPROVED** [review]

**Pull request merged**